src/utils/async.ts

```diff
@@ -0,0 +1,5 @@
+function isPromise(object: any): boolean {

```

Comment:
can change
```
function isPromise(object: any): boolean {
```
to
```
function isPromise(obj: any): obj is Promise {
```

then you will get auto type narrowing, so inside if blocks where it's true, no need to cast

Reply:
learn something everyday :)

Reply:
Good stuff coming in 3.7 re: type narrowing :)
https://github.com/microsoft/TypeScript/pull/32695